### PR TITLE
#2923 - Fixed redirect on new Item version creation

### LINKED ISSUE #2923: The Item version input form is not displayed in DSPace 8, preventing the creation of a new version of an item.
**Describe the bug**
In DSpace 8 Sandbox I try to create a new version for different items (with or without file, with relations and without), results are differents (sometimes UI is freezing, sometimes I only get a confirmation that a version has been created. However, I never get to the new submission form for the new version as describe in the document: https://wiki.lyrasis.org/display/DSDOC8x/Versioned+Item

**To Reproduce**
1. Connect as administrator
2. Find and open an item in the repository
3. Clic on the new version button in the upper left of the page image
4. Add text to describe the version and then clic on Create
5. Want a few seconds, a confirmation message appears: New version has been created with version number 2
6. But, the new version submission form is not displayed.

**Expected behavior**
Should work in described in the documentation  https://wiki.lyrasis.org/display/DSDOC8x/Versioned+Item

### PR DESCRIPTION
**Pull request by Simone-Ramundi**
## References

* Fixes #2923

## Description
Fixed user redirect after creating a new version of an item. Replaced the old method `this.getIDHref()` with `this.searchData.getSearchByHref()`.

## Instructions for Reviewers

List of changes in this PR:
* Implemented new solution inside method `findByItem()`;
* Changed tests.

**To Test**:

- Connect as administrator;
- Find and open an item in the repository;
- Click on the new version button in the upper left of the page;
- Add text to describe the version and then click on Create;
- Want a few seconds, a confirmation message appears: New version has been created with version number 2;
- The new version submission form is displayed.

## Checklist

- [X] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [X] My PR passes [ESLint](https://eslint.org/) validation using `yarn lint`
- [X] My PR doesn't introduce circular dependencies (verified via `yarn check-circ-deps`)
- [X] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [X] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [X] If my PR includes new libraries/dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [X] If my PR includes new features or configurations, I've provided basic technical documentation in the PR itself.
- [X] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

**Review by tdonohue — APPROVED** [review]
👍 Thanks @Simone-Ramundi !  I've tested this today and verified this fixes the bug. Versioning is now redirecting properly to the submission form

**Comment by pilasou** [comment]
Hi @Simone-Ramundi , I have tested the fix and thought it worked for a standard Item (I am redirected to the form for the new version), when trying to create a new version of an item with relations, I get an error message (for reference: I used a standard Item part the dataset provided by DSpace: http://localhost:4000/entities/publication/1974d7d3-9648-4dbe-b083-56374f5314e4): 

Error from chrome dev tool: 
image

DSpace backend log error: 
[pr3069-backendlog-error.txt](https://github.com/DSpace/dspace-angular/files/15419862/pr3069-backendlog-error.txt)

**Comment by tdonohue** [comment]
@pilasou and @Simone-Ramundi :  I was able to reproduce @pilasou 's issue as well.  After further investigation, I believe the bug that @pilasou has found is **directly related** to my open PR #9509.  I think it *should be fixed* by that PR, but I'm going to test it further today.

So, I'm going to turn PR's report into a *new issue ticket* and claim it for myself.  I think I should be able to fix it in #9509.

In summary, I think this PR from @Simone-Ramundi **was successful** in fixing the frontend bug. However, the frontend fix allowed @pilasou to discover a backend bug that _should be fixable_ in #9509.

Thanks @pilasou for your additional testing here!
